tutorpicasso/commands/run_extra_commands.py

```diff
@@ -0,0 +1,142 @@
+import re
+import subprocess
+
+# Was necessary to use this for compatibility with Python 3.8
+from typing import Any, List
+
+import click
+from tutor import config as tutor_config
+
+COMMAND_CHAINING_OPERATORS = ["&&", "&", "||", "|", ";"]
+
+
+@click.command(name="run-extra-commands", help="Run tutor commands")
+def run_extra_commands() -> None:
+    """
+    This command runs tutor commands defined in PICASSO_EXTRA_COMMANDS
+    """
+    tutor_root = (
+        subprocess.check_output("tutor config printroot", shell=True)
+        .decode("utf-8")
+        .strip()
+    )
+    config: Any = tutor_config.load(tutor_root)
+    picasso_extra_commands: Any = config.get("PICASSO_EXTRA_COMMANDS", None)
+    if picasso_extra_commands is not None:
+        validate_commands(picasso_extra_commands)
+        for command in picasso_extra_commands:
+            run_command(command)
+
+
+def validate_commands(commands: Any) -> None:
+    """
+    Takes all the extra commands sent through config.yml and verifies that
+    all the commands are correct before executing them
+
+    Args:
+        commands (list[str] | None): The commands sent through PICASSO_EXTRA_COMMANDS in config.yml
+    """
+    splitted_commands = [
+        split_string(command, COMMAND_CHAINING_OPERATORS) for command in commands
+    ]
+    flat_commands_array: List[str] = sum(splitted_commands, [])
+
+    invalid_commands = []
+    misspelled_commands = []
+    for command in flat_commands_array:
+        if "tutor" not in command.lower():
+            if find_tutor_misspelled(command):
+                misspelled_commands.append(command)
+            else:
+                invalid_commands.append(command)
+
+        if invalid_commands or misspelled_commands:
+            error_message = (
+                f"Found some issues with the commands:\n\n"
+                f"{'=> Invalid commands: ' if invalid_commands else ''}"
+                f"{', '.join(invalid_commands) if invalid_commands else ''}\n"
+                f"{'=> Misspelled commands: ' if misspelled_commands else ''}"
+                f"{', '.join(misspelled_commands) if misspelled_commands else ''}\n"
+                f"Take a look at the official Tutor commands: "
+                f"https://docs.tutor.edly.io/reference/cli/index.html"
+            )
+            raise click.ClickException(error_message)
+
+
+def run_command(command: str) -> None:
+    """
+    Run an extra command.
+
+    This method runs the extra command provided.
+
+    Args:
+            command (str): Tutor command.
+    """
+    try:
+        with subprocess.Popen(
+            command,
+            shell=True,
+            executable="/bin/bash",
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            text=True,
+        ) as process:
+
+            # It is sent a 'y' to say 'yes' on overriding the existing folders
```

Comment:
Could you please clarify what you mean by this comment?

Reply:
The comment concerns the following line, which sends a y or yes if the tutor command needs input.

Reply:
I bring the code from the distro plugin. I am unsure if that is strictly necessary. I will try to remove that.

Reply:
Sorry, yes. We had that line because in the distro if you install a private package that is already in the requirements, the distro asks you if you want to override it. So, suppose you had an enable-private-package command inside the run-extra-commands. In that case, you will need the y or yes answer to override private packages if they are inside the extra-commands.

But this is no longer necessary because I handle that here: https://github.com/eduNEXT/tutor-contrib-picasso/blob/mfmz/simple-picasso/tutorpicasso/commands/enable_private_packages.py#L49

Reply:
I already remove the input 'y'. And It worked as expected.